src/receive_imf.rs

```diff
@@ -1037,9 +1037,13 @@ async fn add_parts(
         // the mail is on the IMAP server, probably it is also delivered.
         // We cannot recreate other states (read, error).
         state = MessageState::OutDelivered;
-        to_id = to_ids.first().copied().unwrap_or_default();
+        to_id = to_ids.first().copied().unwrap_or(ContactId::SELF);
 
-        let self_sent = to_ids.len() == 1 && to_ids.contains(&ContactId::SELF);
+        // Older Delta Chat versions with core <=1.152.2 only accepted
+        // self-sent messages in Saved Messages with own address in the `To` field.
+        // New Delta Chat versions may use empty `To` field
+        // with only a single `hidden-recipients` group in this case.
+        let self_sent = to_ids.is_empty() || (to_ids.len() == 1 && to_id == ContactId::SELF);
```

Comment:
Can be rewritten as `to_ids.len() <= 1 && to_id == ContactId::SELF`? `to_id` defaults to `ContactId::SELF` now